docs/2-guides/2-island-architecture/README.md

```diff
@@ -0,0 +1,302 @@
+
+Last but not least, the `children` prop, which is the technical name for all children passed to a React component. (`<Parent children={<Child />} />` is the same as `<Parent><Child /></Parent>`.)
+
+The `<Island />` component can take children, but its behavior depends on its `clientOnly` prop.
+
+In default mode (without `clientOnly`), the children are rendered on the server and sent to the client, as children of your island component. The children will not be made interactive.
+
+This behavior enables components like accordions, where the `<Island />` is not a leaf of the component tree.
+
+[Schema of the accordion component](./accordion.svg)
+
+Such a component can be implemented as follows:
+
+```tsx
+// Accordion.client.tsx
+import type { ReactNode } from "react";
+
+export default function Accordion({ children }: { children: ReactNode }) {
+  // The accordion can be opened and closed
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <div>
+      <button onClick={() => setIsOpen(!isOpen)}>{isOpen ? "Close" : "Open"} accordion</button>
+      <div style={{ display: isOpen ? "block" : "none" }}>
+        {/* Children will be inserted here: */}
+        {children}
+      </div>
+    </div>
+  );
+}
+```
+
+And on the server:
+
+```tsx
+// default.server.tsx
+import { Island, jahiaComponent } from "@jahia/javascript-modules-library";
+import Accordion from "./Accordion.client.tsx";
+
+jahiaComponent(
+  {
+    componentType: "view",
+    nodeType: "hydrogen:example",
+  },
+  () => (
+    <article>
+      <Island component={Accordion}>
+        <p>I'm rendered on the server!</p>
+      </Island>
+    </article>
+  ),
+);
+```
+
+A few things to note:
+
+- The `{children}` insertion point must always be there. If you want to hide the children of your component, use CSS instead of a JS condition. Otherwise, they will not be sent to the client and your component will appear to have no children.
+- Children will be wrapped in a `jsm-children` element. This should not affect your code most of the time, but don't use the `>` CSS selector to target children of your component.
+
+In `clientOnly` mode, the children of an island will not be used as children of your island component. Instead, they will be rendered on the server and used as a placeholder until the client component is loaded.
+
+```tsx
+// default.server.tsx
+import { Island, jahiaComponent } from "@jahia/javascript-modules-library";
+import Map from "./Map.client.tsx";
+
+jahiaComponent(
+  {
+    componentType: "view",
+    nodeType: "hydrogen:example",
+  },
+  () => (
+    <article>
+      <Island component={Map}>
+        {/* Placeholder until <Map /> has loaded */}
+        <p>The map is loading...</p>
+      </Island>
+    </article>
+  ),
+);
+```
+
+This is a good UX practice to tell show users that your site is currently loading instead of leaving an empty space. It can also prevent [layout shifts](https://web.dev/articles/cls) when the component finally loads.
```

Comment:
```suggestion
This is a good UX practice to tell users that your site is currently loading instead of leaving an empty space. It can also prevent [layout shifts](https://web.dev/articles/cls) when the component finally loads.
```